examples/mnist/train_mnist_data_parallel.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import argparse
+
+import chainer
+import chainer.links as L
+from chainer import training
+from chainer.training import extensions
+
+import train_mnist
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Chainer example: MNIST')
+    parser.add_argument('--batchsize', '-b', type=int, default=400,
+                        help='Number of images in each mini batch')
+    parser.add_argument('--epoch', '-e', type=int, default=20,
+                        help='Number of sweeps over the dataset to train')
+    parser.add_argument('--gpu0', '-g', type=int, default=0,
+                        help='First GPU ID')
+    parser.add_argument('--gpu1', '-G', type=int, default=1,
+                        help='Second GPU ID')
+    parser.add_argument('--out', '-o', default='result_parallel',
+                        help='Directory to output the result')
+    parser.add_argument('--resume', '-r', default='',
+                        help='Resume the training from snapshot')
+    parser.add_argument('--unit', '-u', type=int, default=1000,
+                        help='Number of units')
+    args = parser.parse_args()
+
+    print('GPU: {}, {}'.format(args.gpu0, args.gpu1))
+    print('# unit: {}'.format(args.unit))
+    print('# Minibatch-size: {}'.format(args.batchsize))
+    print('# epoch: {}'.format(args.epoch))
+    print('')
+
+    # See train_mnist.py for the meaning of these lines
+    model = L.Classifier(train_mnist.MLP(784, args.unit, 10))
+
+    # Make a specified GPU current
+    chainer.cuda.get_device(args.gpu0).use()
+
+    # Setup an optimizer
+    optimizer = chainer.optimizers.Adam()
+    optimizer.setup(model)
+
+    # Load the MNIST dataset
+    train, test = chainer.datasets.get_mnist()
+
+    train_iter = chainer.iterators.ShuffledIterator(train, args.batchsize)
+    test_iter = chainer.iterators.SequentialIterator(test, args.batchsize,
+                                                     repeat=False)
+
+    # Set up a trainer
+    updater = training.ParallelUpdater(
+        train_iter,
+        optimizer,
+        devices={'main': args.gpu0, 'second': args.gpu1},
+    )
+    trainer = training.Trainer(updater, (args.epoch, 'epoch'), out=args.out)
+
+    # Evaluate the model with the test dataset for each epoch
+    trainer.extend(extensions.Evaluator(test_iter, model, device=args.gpu0))
+
+    # Dump a computational graph from 'loss' variable at the first iteration
+    trainer.extend(extensions.dump_graph('main/loss'))
+
+    # Take a snapshot at each epoch
+    trainer.extend(extensions.snapshot())
+
+    # Write a log of evaluation statistics for each epoch
+    trainer.extend(extensions.LogReport())
+
+    # Print selected entries of the log to stdout
+    # Entries prefixed by 'validation' are computed by the Evaluator extension
```

Comment:
Same as train_mnist, explain how to derive the entries. Explain what 'main' refers to.